### DISCUSSION TITLE
Question regarding re-using mean fields in optimization

### DISCUSSION THREAD

**Discussion by uttamcadambi07** [mdolab/dafoam#444] [General]
Hi,

I have a question about the DAFoam and IPOPT.

Specifically the fieldInversion case.

I see from the log that there exists the line "Previous solution time XXXXX found and deleted". 
I have two questions:
1. Does the optimizer use the beta field from the previous iteration?
2. Does the optimizer use the mean fields i.e. U, p, nut., nuTilda (for SA model for example) from the previous iteration or does the primal solver start from uniform fields?
I am asking because if it starts from a uniform field each time the IPOPT optimizer is called with a beta field from the previous optimization iteration, then the number of primal solver calls becomes too high. 

I am a little confused with the order of execution so please help me understand it better.

**Comment by friedenhe** [comment]
1. No. The optimizer decides the beta field. In each iteration, the optimizer assigns new beta values to the flow field before the primal runs.
2. The U, p, phi, nut, variables are saved in the memory, so the primal solution for the nth iteration will use the U, p, phi from the (n-1)th iteration as the initial condition. They are NOT uniform.
